sharktank/sharktank/types/tensors.py

```diff
@@ -1385,16 +1387,19 @@ def __init__(
         will be replicated that many times.
         """
         if devices is None:
-            num_shards = shard_count if isinstance(ts, torch.Tensor) else len(ts)
+            num_shards = len(ts) if isinstance(ts, list) else shard_count
             devices = tuple(range(num_shards))
 
-        if isinstance(ts, torch.Tensor):
+        if isinstance(ts, (torch.Tensor, QuantizedTensor)):
```

Comment:
We should make this a check for whether its any type of tensor. `torch.Tensor` or sharktank tensor type. We could also just check whether its a list.